lib/improver/wind_gust_diagnostic.py

```diff
@@ -201,18 +201,30 @@ def process(self, cube_gust, cube_ws):
                    ' {0:s} {1:s}.'.format(perc_coord_gust.name(),
                                           perc_coord_ws.name()))
             raise ValueError(msg)
-        # Add metadata to both cubes
+
+        # Check times are compatible.
+        msg = ('Could not match time coordinate')
+        try:
+            wg_time = req_cube_gust.coord('time')
+            ws_time = req_cube_ws.coord('time')
+            for i, point in enumerate(wg_time.points):
+                if point != ws_time.points[i]:
+                    if ws_time.points[i] < wg_time.bounds[i][0]:
+                        raise ValueError(msg)
+                    elif ws_time.points[i] > wg_time.bounds[i][1]:
+                        raise ValueError(msg)
+        except:
+            raise ValueError(msg)
```

Comment:
I'm not sure about this. I added this in because I didn't want to have to put in a lot of if statements to check lots of different things like whether cubes have time coordinate or have bounds so there are likely to be a few Iris / UserValue exceptions. If you are not happy with a general except it would probably be better to rewrite this.

Reply:
I see what you mean.....could the multiple Iris / User value exceptions just be caught in the same line i.e.

         except (IrisException, UserValueException) as error;
            raise.ValueError(msg)

Also reduce the if/elif to:

        if ws_time.points[i] < wg_time.bounds[i][0] or ws_time.points[i] > wg_time.bounds[i][1]:
            raise ValueError(msg)

Reply:
I've now rewritten it now so that it doesn't use except. Please see recent changes.

I do not see the point of changing the if/elif as the two are equivalent.